pennylane/resources/factorization.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for two-electron tensor factorization.
+"""
+from pennylane import numpy as np
+
+
+def factorize(two, tol):
+    r"""Return the double-factorized form of a two-electron tensor.
+
+    The two-electron tensor :math:`V`, in the chemist notation, is first factorized in terms of
+    symmetric matrices :math:`L^{(r)}` such that
+    :math:`V_{ijkl} = \sum_r^R L_{ij}^{(r)} L_{kl}^{(r) T}`. The rank :math:`R` is determined by a
+    threshold error. Then, each matrix :math:`L^{(r)}` is diagonalized and its eigenvalues (and
+    corresponding eigenvectors) are truncated with the same threshold error.
+
+    Args:
+        two (array[array[float]]): the two-electron repulsion tensor in the molecular orbital basis
+            arranged in chemist notation
```

Comment:
Not every user may know what chemist notation means. Maybe we can add a link? I found this one http://vergil.chemistry.gatech.edu/notes/permsymm/permsymm.pdf

Reply:
Added.